### PR TITLE
Rename oneHotLabels to probabilities.

### PR DESCRIPTION
**Pull request by xiejw**

### CODE REVIEW

File: Sources/DeepLearning/Loss.swift [hunk 1]
```diff
@@ -54,18 +54,32 @@ func _vjpSoftmaxCrossEntropy<Scalar: TensorFlowFloatingPoint>(
...
+///   - logits: One-hot encoded outputs from a neural network.
+///   - oneHotLabels: One-hot encoded values that correspond to the correct output.
+@differentiable(wrt: logits, vjp: _vjpSoftmaxCrossEntropy)
+public func softmaxCrossEntropy<Scalar: TensorFlowFloatingPoint>(
+    logits: Tensor<Scalar>, probabilities: Tensor<Scalar>
+) -> Tensor<Scalar> {
+    return Raw.softmaxCrossEntropyWithLogits(features: logits, labels: probabilities).loss.mean()
+}
+
+
```
Comment (rxwei) [inline Sources/DeepLearning/Loss.swift]: Rename extra line.

File: Sources/DeepLearning/Loss.swift [hunk 2]
```diff
@@ -54,18 +54,32 @@ func _vjpSoftmaxCrossEntropy<Scalar: TensorFlowFloatingPoint>(
...
 ) -> Tensor<Scalar> {
-    return Raw.softmaxCrossEntropyWithLogits(features: logits, labels: oneHotLabels).loss.mean()
+    return softmaxCrossEntropy(logits: logits, probabilities: oneHotLabels)
 }
 
+/// Computes the softmax cross entropy (categorical cross entropy) between logits and labels.
+///
+/// - Parameters:
+///   - logits: One-hot encoded outputs from a neural network.
+///   - oneHotLabels: One-hot encoded values that correspond to the correct output.
```
Comment (marcrasi) [inline Sources/DeepLearning/Loss.swift]: fix this parameter documentation

File: Sources/DeepLearning/Loss.swift [hunk 3]
```diff
@@ -54,18 +54,32 @@ func _vjpSoftmaxCrossEntropy<Scalar: TensorFlowFloatingPoint>(
 /// - Parameters:
 ///   - logits: One-hot encoded outputs from a neural network.
 ///   - oneHotLabels: One-hot encoded values that correspond to the correct output.
-@differentiable(wrt: logits, vjp: _vjpSoftmaxCrossEntropy)
+@available(*, deprecated, renamed: "softmaxCrossEntropy(logits:probabilities:)")
```
Comment (marcrasi) [inline Sources/DeepLearning/Loss.swift]: Should we just delete this? We provide no backwards-compatibility guarantees right now, and I think we should aggressively delete stuff to keep things clean.

Comment (rxwei) [reply Sources/DeepLearning/Loss.swift]: Sure, works for me.

File: Sources/DeepLearning/Loss.swift (hunk 2 above)
Comment (xiejw) [reply Sources/DeepLearning/Loss.swift]: Done.

File: Sources/DeepLearning/Loss.swift (hunk 1 above)
Comment (xiejw) [reply Sources/DeepLearning/Loss.swift]: Done.

File: Sources/DeepLearning/Loss.swift (hunk 3 above)
Comment (xiejw) [reply Sources/DeepLearning/Loss.swift]: Deleted.
